src/stcal/resample/resample.py

```diff
@@ -1109,38 +1074,13 @@ def finalize_resample_variance(self, output_model):
             warnings.filterwarnings("ignore", "invalid value*", RuntimeWarning)
             warnings.filterwarnings("ignore", "divide by zero*", RuntimeWarning)
 
-            output_variance = (
-                self._var_rnoise_wsum / (self._var_rnoise_weight *
-                                         self._var_rnoise_weight)
-            ).astype(
-                dtype=self.output_array_types["var_rnoise"]
-            )
-            output_model["var_rnoise"] = output_variance
-
-            output_variance = (
-                self._var_poisson_wsum / (self._var_poisson_weight *
-                                          self._var_poisson_weight)
-            ).astype(
-                dtype=self.output_array_types["var_poisson"]
-            )
-            output_model["var_poisson"] = output_variance
-
-            output_variance = (
-                self._var_flat_wsum / (self._var_flat_weight *
-                                       self._var_flat_weight)
-            ).astype(
-                dtype=self.output_array_types["var_flat"]
-            )
-            output_model["var_flat"] = output_variance
-
-            del (
-                self._var_rnoise_wsum,
-                self._var_poisson_wsum,
-                self._var_flat_wsum,
-                self._var_rnoise_weight,
-                self._var_poisson_weight,
-                self._var_flat_weight,
-            )
+            for varname in self._enable_var:
+                varwsum = self.variance_info[varname]['wsum']
+                weight = self.variance_info[varname]['wt']
+                output_model[varname] = (varwsum / (weight * weight)).astype(
+                    dtype=self.output_array_types[varname]
+                )
+            del self.variance_info
```

Comment:
I don't like when class attributes can disappear (especially public ones). Maybe iterate over dict items and delete those? and also make this a private attribute?

Reply:
Thanks.  I've renamed this to _variance_info.  I didn't delete all of the dictionary entries individually, though I could---I'll admit it's unclear to me when I'm happy to let the garbage collector do its collection on its own vs. when I want to remove things explicitly.